### PR TITLE
storage/mysql: Fix 2 mysql TEXT COLUMNS bugs

### PR DESCRIPTION
**Pull request by rjobanp**
### Motivation

  * This PR fixes a recognized bug. Fixes https://github.com/MaterializeInc/materialize/issues/25884 and Fixes https://github.com/MaterializeInc/materialize/issues/25883

Fix 2 issues related to MySQL `TEXT COLUMNS` and `IGNORE COLUMNS` purification/planning.

### Tips for reviewer

The first issue was when encountering column names with special characters -- we were inadvertently storing a postgres-style-escaped string for the column name in our source connection options which later led to errors when verifying the schema in the source operator.

The second issue was ignoring any column references that didn't match -- this change just propagates those up as an error to the user during sql purification.

The 2 commits correspond to each of the issues fixed.

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [x] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [x] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [x] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [x] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

**Review by mjibson — COMMENTED** [review]
Adapter parts lgtm.

File: src/storage-types/src/sources/mysql.rs [hunk 1]
```diff
@@ -63,6 +64,37 @@ impl RustType<ProtoMySqlColumnRef> for MySqlColumnRef {
...
+    fn try_from(item: UnresolvedItemName) -> Result<Self, Self::Error> {
+        let mut iter = item.0.into_iter();
+        match (iter.next(), iter.next(), iter.next()) {
+            (Some(schema_name), Some(table_name), Some(column_name)) => Ok(MySqlColumnRef {
+                // We need to be careful not to accidentally introduce postgres/mz-style
+                // quoting in the schema_name, table_name, or column_name fields, which
+                // happens if naively using the to_string()/fmt method on the Ident values.
+                schema_name: schema_name.into_string(),
+                table_name: table_name.into_string(),
+                column_name: column_name.into_string(),
```
Comment (guswynn) [inline src/storage-types/src/sources/mysql.rs:88]: wild bug, omg

**Review by guswynn — APPROVED** [review]
v nice!

File: src/storage-types/src/sources/mysql.rs (hunk 1 above)
Comment (rjobanp) [reply src/storage-types/src/sources/mysql.rs:88]: tell me about it haha. It took me a while for my brain to notice `into_string` and `to_string` were different 😅

**Pull request merged**